src/executor.rs

```diff
@@ -25,6 +25,7 @@ pub mod asynch {
     }
 
     #[derive(Clone, Debug)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
```

Comment:
How does `derive(defmt::Format)` behave when `B` and/or `T` **do not** implement `defmt::Format`? I do hope it will *not* require those to implement `defmt::Format` and expands to a conditional implementation along the lines of:
```rust
impl defmt::Format for Blocking<B, T>
where
    B: defmt::Format,
    T: defmt::Format,
{
...
}
```

... if you could confirm?

Reply:
It works the same as ``Debug``, so I think in this case it will error out if they dont already implement ``defmt::Format``, I can remove the ``derive(defmt::Format`` or tell ``defmt`` to use ``Debug2Format`` or stuff like that.
There is [more info in their book](https://defmt.ferrous-systems.com/format.html)

Reply:
derive macros don't modify the item they are attached to, so I don't believe it would add any required bounds to `B` or `T` except in the impl of the trait.

You can confirm with `cargo expand` and you can even add a test case if you want, try instantiating a `Blockin` with a `B` and a `T` that don't implement defmt.